src/ray/core_worker/core_worker.cc

```diff
@@ -1010,6 +1010,22 @@ CoreWorker::CoreWorker(CoreWorkerOptions options, const WorkerID &worker_id)
 CoreWorker::~CoreWorker() { RAY_LOG(INFO) << "Core worker is destructed"; }
 
 void CoreWorker::Shutdown() {
+  // For actors, perform cleanup before shutdown starts
+  {
+    absl::MutexLock lock(&mutex_);
```

Comment:
why is this guarded with the mutex rather than the `is_shutdown` atomic guard below?